src/platform_impl/linux/wayland/window/state.rs

```diff
@@ -358,13 +353,27 @@ impl WindowState {
                 .unwrap_or(new_size.height);
         }
 
-        // XXX Set the configure before doing a resize.
-        self.last_configure = Some(configure);
+        let new_state = configure.state;
+        let old_state = self
+            .last_configure
+            .as_ref()
+            .map(|configure| configure.state)
+            .unwrap_or(XdgWindowState::empty());
+
+        let state_change_requires_resize = !new_state
+            .symmetric_difference(old_state)
+            .difference(XdgWindowState::ACTIVATED | XdgWindowState::SUSPENDED)
+            .is_empty();
```

Comment:
This took a little bit to understand, should it have a comment explaining that _any_ change in state will trigger a resize _except these two_?

Maybe this should be an opt-in anyway?

Reply:
I mean the code is self documenting if you account for variable name. I had a comment wrt that, but the variable name clearly tells that change requires resize when anything other that activated/suspended changed.

Reply:
It takes a bit of processing to figure out through three function calls and a negation operator that this is "if there's any change in state that _is not_ `ACTIVATED|SUSPENDED`".

Reply:
Hm, maybe, for me it's just all set operations, I've added comment, but that's about it.